### PR TITLE
[WIP] Fullscreen App Library on Landscape

### PR DESCRIPTION
**Pull request by PoomSmart**

### CODE REVIEW

**Comment by PoomSmart** [comment]
We may hook:

```
%hook SBHLibraryCategoriesRootFolder

- (id)initWithDisplayName:(NSString *)displayName maxListCount:(NSUInteger)maxListCount listGridSize:(SBHIconGridSize)listGridSize iconGridSizeClassSizes:(void *)iconGridSizeClassSizes {
	return %orig(displayName, maxListCount, (SBHIconGridSize){ .rows = 0xFFFF, .columns = 8 }, iconGridSizeClassSizes);
}

%end
```

The problem is that, even we set the columns depending on the interface orientation, the first value assigned will be used throughout the experiences.

**Comment by PoomSmart** [comment]
Adding this hook will make the landscape layout nearly perfect, but SpringBoard will crash when an application outside the certain bounds is tapped.

```
%hook SBHLibraryCategoriesRootFolder

- (id)initWithDisplayName:(NSString *)displayName maxListCount:(NSUInteger)maxListCount listGridSize:(SBHIconGridSize)listGridSize iconGridSizeClassSizes:(void *)iconGridSizeClassSizes {
	return %orig(displayName, maxListCount, (SBHIconGridSize){ .rows = 0xFFFF, .columns = 8 }, iconGridSizeClassSizes);
}

%end
```

**Comment by tomaszpoliszuk** [comment]
Seems to be breaking landscape view, doesn't apply to portrait and can't access app library when upside down after this changes.

Simulator Screen Shot - iPad Pro (11-inch) (2nd generation) - 2021-06-09 at 20 35 10

Simulator Screen Shot - iPad Pro (11-inch) (2nd generation) - 2021-06-09 at 20 33 55

File: AppLibraryEnabler.xm [hunk 1]
```diff
@@ -127,26 +139,40 @@
...
-//	}
-//	%end
-
-extern "C" bool _os_feature_enabled_impl(const char *domain, const char *feature);
-%hookf(bool, _os_feature_enabled_impl, const char *domain, const char *feature) {
-	if (strcmp(domain, "SpringBoard") == 0 && strcmp(feature, "Dewey") == 0) {
-		return true;
-	}
-	return %orig;
-}
```
Comment (PoomSmart) [inline AppLibraryEnabler.xm]: I find that this feature flag is already enabled on iPad, thus no longer needed. You may want to verify if this is the case for older iPad models.

**Comment by tomaszpoliszuk** [comment]
Yesterday i was testing and maybe this part will be helpful:

```
typedef struct SBHIconGridSize {
	unsigned short columns;
	unsigned short rows;
} SBHIconGridSize;

typedef struct SBHIconGridSizeClassSizes {
	SBHIconGridSize small;
	SBHIconGridSize medium;
	SBHIconGridSize large;
	SBHIconGridSize extraLarge;
} SBHIconGridSizeClassSizes;

%hook SBHLibraryCategoriesRootFolder
-(id)initWithDisplayName:(id)displayName maxListCount:(unsigned long long)arg2 listGridSize:(SBHIconGridSize)iconGridSize iconGridSizeClassSizes:(SBHIconGridSizeClassSizes)gridSizeClassSizes {
	if ( [displayName isEqualToString:@"Categories Folder"] ) {
//		SBHIconGridSize newIconGridSize = iconGridSize;
//		newIconGridSize.columns = 8;

		SBHIconGridSizeClassSizes newGridSizeClassSizes = gridSizeClassSizes;
		newGridSizeClassSizes.small.columns = 1;
		newGridSizeClassSizes.small.rows = 1;
		return %orig( displayName, arg2, iconGridSize, newGridSizeClassSizes );
//		return %orig( displayName, arg2, newIconGridSize, newGridSizeClassSizes );
	}
	return %orig( displayName, arg2, iconGridSize, gridSizeClassSizes );
}
%end
```

It was setting same spacing between categories but it was too small (categories were stacking on each other) and _SBHLibraryPodIconListView frame was too small (but it was in sync with their spacing).

File: AppLibraryEnabler.xm [hunk 2]
```diff
@@ -136,22 +132,6 @@ typedef struct SBHIconGridSize {
...
 	searchResultsContainerView.frame = selfFrame;
 }
-- (void)_layoutSearchViews {
-	%orig;
-	MTMaterialView *searchBackdropView = [self valueForKey:@"_searchBackdropView"];
-
-	CGFloat width = [[UIScreen mainScreen] bounds].size.width;
-	CGFloat height = [[UIScreen mainScreen] bounds].size.height;
-
-	CGRect fullScreenFrame = CGRectMake(
```
Comment (tomaszpoliszuk) [inline AppLibraryEnabler.xm:114]: This part will be needed for application list (only height and top offset part) - without it overlay in this view is cut out on top and bottom.

File: AppLibraryEnabler.xm [hunk 3]
```diff
@@ -136,22 +132,6 @@ typedef struct SBHIconGridSize {
 	contentContainerView.frame = selfFrame;
 	searchResultsContainerView.frame = selfFrame;
 }
-- (void)_layoutSearchViews {
```
Comment (tomaszpoliszuk) [inline AppLibraryEnabler.xm:107]: This part will be needed for application list (only height and top offset part) - without it overlay in this view is cut out on top and bottom.

**Pull request closed without merging**